File: TrainingProject/Assignments/FrontEnd/Source/Frontend/src/Models/Theatre.ts
// [Name]
//       ,[Street]
//       ,[City]
//       ,[State]
//       ,[Country]
//       ,[CreateDate]
//       ,[UpdateDate]
//       ,[CreatedByID]
//       ,[UpdateByID]

export interface Theatre{
    name:string,
    street:string,
    city:string,
    state:string,
    country:string,
    uid:string,
    minprice:number
    facilities:number[]
}

export interface TheatreView{
    id:number
    name:string,
    street:string,
    city:string,
    state:string,
    country:string,
    uid:string,
    facilities:Facility[]
}

export interface Facility{
    id:number
    name:string,
    url:string
}

File: TrainingProject/Assignments/API/Source/API/API/Models/Theatre.cs
using System;
using System.Collections.Generic;

namespace API.Models;

public partial class Theatre
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string Street { get; set; } = null!;

    public string City { get; set; } = null!;

    public string State { get; set; } = null!;

    public string Country { get; set; } = null!;

    public DateTime? CreateDate { get; set; }

    public DateTime? UpdateDate { get; set; }

    public int? CreatedById { get; set; }

    public int? UpdateById { get; set; }

    public decimal MinTicketPrice { get; set; }

    public bool? Status { get; set; }

    public virtual User? CreatedBy { get; set; }

    public virtual ICollection<Screen> Screens { get; set; } = new List<Screen>();

    public virtual ICollection<TheatreFacility> TheatreFacilities { get; set; } = new List<TheatreFacility>();

    public virtual User? UpdateBy { get; set; }
}
